js/index.js

```diff
@@ -1,7 +1,364 @@
+const User_List = document.getElementById('user-list');
+let USER_COUNT =0;
+
+// 해당 서버에 데이터 가져와 처리
+function Response_Api(){
+  fetch('https://js-todo-list-9ca3a.df.r.appspot.com/api/users')
+      .then((response) => response.json())
+      .then((myJson)=> {
+        for(let i=0;i<myJson.length;i++)
+          Get_Users(myJson,i);
+        Click_User();
+        document.querySelector('.user-create-button').addEventListener('click',Make_User);
+        document.querySelector('.user-delete-button').addEventListener('click',Delete_User);
+      })
+};
+
+//해당 데이터 내용을 가진 name 추가
+function Get_Users(data,i){
+  const $user_list = document.getElementById('user-list');
+  const button = `<button class="ripple" id=${++USER_COUNT}>${data[i].name}</button>`
+  const appRender = () => {
+    $user_list.innerHTML += button;
+  }
+  appRender();
+}
+
+// 클릭시 생성 혹은 삭제 핸들러 실행
+function Make_User(){
+  document.querySelector('.user-create-button').addEventListener('click',onUserCreateHandler)
+}
+function Delete_User(){
+  document.querySelector('.user-delete-button').addEventListener('click',onUserDeleteHandler)
+}
+
+//추가한 이름을 브라우저 및 서버에 적용
 const onUserCreateHandler = () => {
   const userName = prompt("추가하고 싶은 이름을 입력해주세요.");
+  // User의 이름은 최소 2글자 이상체크
+  let User_Enough_Length = userName.length>1 ? true : false;
+  if(User_Enough_Length) {
+    const button = document.createElement('button');
+    button.innerText = userName;
+    button.classList.add('ripple');
+    button.setAttribute('id',++USER_COUNT);
+    User_List.insertBefore(button, document.querySelector('user-create-button'));
+    POST_USER(userName);
+
+    //추가된 만큼 핸들러 다시 추가
+    USER_COUNT++;
+    Click_User();
+  }
+}
+
+//삭제한 이름을 가진 데이터를 브라우저 및 서버에 적용
+const onUserDeleteHandler = ()=>{
+  const userName = prompt("삭제하고 싶은 이름을 입력해주세요.");
+  document.querySelectorAll('#user-list > button').forEach(x=>{
+    if(x.innerText === userName){
+      if(x.classList.contains('user-create-button')==true || x.classList.contains('user-delete-button')==true){
```

Comment:
```suggestion
      if(x.classList.contains('user-create-button') || x.classList.contains('user-delete-button')){
```

```ture```일 경우 조건문 내부로 들어가게 하시려면 위와 같이 작성하셔도 됩니다! 덧붙여서 false조건일 경우에는 위와 같이 작성한 코드에 논리 연산자(!)를 붙여서 표현 가능합니다!

```js
      if(!x.classList.contains('user-create-button') || !x.classList.contains('user-delete-button')){
```